### PR TITLE
KubeflowJob: Commonize OrderedReplicaTypes and PriorityClass

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
In the `KubeflowJob`, I commonized the `OrderedReplicaTypes()` and the `PriorityClass()` functions as discussed in https://github.com/kubernetes-sigs/kueue/pull/1068#discussion_r1298664579.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller_test.go [hunk 1]
```diff
@@ -167,3 +168,50 @@ func TestCalcPriorityClassName(t *testing.T) {
 		})
 	}
 }
+
+func TestOrderedReplicaTypes(t *testing.T) {
```
Comment (alculquicondor) [inline pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller_test.go:172]: this looks outdated

File: pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go [hunk 2]
```diff
@@ -70,7 +70,7 @@ func (j *KubeflowJob) RunWithPodSetsInfo(podSetInfos []jobframework.PodSetInfo)
 }
 
 func (j *KubeflowJob) RestorePodSetsInfo(podSetInfos []jobframework.PodSetInfo) bool {
-	orderedReplicaTypes := j.KFJobControl.OrderedReplicaTypes(j.KFJobControl.ReplicaSpecs())
+	orderedReplicaTypes := j.OrderedReplicaTypes()
 	changed := false
 	for index, info := range podSetInfos {
 		replicaType := orderedReplicaTypes[index]
```
Comment (alculquicondor) [inline pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go:76]: we need to check if it exists in the spec here, right?

Comment (alculquicondor) [reply pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go:76]: ahh, I see that you are filtering in KubeflowJob

**Review by alculquicondor — COMMENTED** [review]
/lgtm

File: pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller_test.go (hunk 1 above)
Comment (tenzen-y) [reply pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller_test.go:172]: Why do you think this is outdated? I think this test works fine.

File: pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go (hunk 2 above)
Comment (tenzen-y) [reply pkg/controller/jobs/kubeflow/kubeflowjob/kubeflowjob_controller.go:76]: Right.

File: pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller.go [hunk 3]
```diff
@@ -85,33 +85,8 @@ func (j *JobControl) JobStatus() kftraining.JobStatus {
...
-	if j.Spec.RunPolicy.SchedulingPolicy != nil && len(j.Spec.RunPolicy.SchedulingPolicy.PriorityClass) != 0 {
-		return j.Spec.RunPolicy.SchedulingPolicy.PriorityClass
-	} else if m := j.Spec.PyTorchReplicaSpecs[kftraining.PyTorchJobReplicaTypeMaster]; m != nil && len(m.Template.Spec.PriorityClassName) != 0 {
-		return m.Template.Spec.PriorityClassName
-	} else if w := j.Spec.PyTorchReplicaSpecs[kftraining.PyTorchJobReplicaTypeWorker]; w != nil && len(w.Template.Spec.PriorityClassName) != 0 {
-		return w.Template.Spec.PriorityClassName
-	}
-	return ""
+func (j *JobControl) OrderedReplicaTypes() []kftraining.ReplicaType {
+	return []kftraining.ReplicaType{kftraining.PyTorchJobReplicaTypeMaster, kftraining.PyTorchJobReplicaTypeWorker}
```
Comment (kerthcet) [inline pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller.go:89]: One question here: we ordered the types is to 1) make sure the scheduling order of different types or 2) just to list the expected types here, which means we can shuffle the order, like scheduling the worker types first. 

We may want to schedule the workers first which dominates the resource allocation. Thanks!

Comment (tenzen-y) [reply pkg/controller/jobs/kubeflow/jobs/pytorchjob/pytorchjob_controller.go:89]: IIUC, the PyTorchJob replicaspec is map, which means no order between each replica type.
So, I guess that we can not enforce the scheduling order.

**Pull request merged**